### DISCUSSION TITLE
Blacklist whitelist directory levels

### DISCUSSION THREAD

**Discussion by PCBscanner** [netblue30/firejail#6500] [Q&A] [answered]
I'm testing firejail with the PCSX2 emulator. I've modified PCSX2.profile to contain the following lines:
```
noblacklist ${HOME}/.config/PCSX2
noblacklist  ${HOME}/Documents
...
mkdir ${HOME}/.config/PCSX2
mkdir ${HOME}/Documents/Games/PS2
whitelist ${HOME}/.config/PCSX2
whitelist  ${HOME}/Documents/Games/PS2
```
This works fine, and PCSX2 can only access from ../PS2 and below.  However, changing `noblacklist  ${HOME}/Documents` to `noblacklist  ${HOME}/Documents/Games` or any directory below ../Documents fails.

Could someone explain why I am encountering this behaviour? I would've expected that `noblacklist`, `mkdir`, `whitelist` being the same should work, but it does not in my case.

**Comment by rusty-snake** [comment, marked as answer]
https://github.com/netblue30/firejail/wiki/Creating-Profiles#common-mistakes

> **--noblacklist=dirname_or_filename**
> Disable blacklist for this directory or file.

In your case `${HOME}/Documents/Games` is never blacklisted so the noblacklist does not have an effect.

**Reply by PCBscanner** [reply]
I see. So the blacklist to `${HOME}/Documents` was applied in one of the includes, therefore `noblacklist ${HOME}/Documents` removes it, but there's a failsafe still active from the initial blacklist, so the app still cannot access that directory or subdirectories therefore you need to `whitelist ${HOME}/Documents/path/to/dir`. Thanks @rusty-snake.
